src/app/shared/services/app.service.ts

```diff
@@ -638,6 +650,25 @@ export class NSSService {
             .get(this.configSettings.gageStatsBaseURL + this.configSettings.stationsURL + "?stationTypes=" + id + "?agencies=" + id2)
     }
 
+    // get stations by text search
+    public searchStations(term: string, id: Array<number>, params?: string) {
+        let url = ''
+        let url2 = ''
+        let url3 = ''
+        if (term) {
+            url = "?filterText=" + term
+        }
+        if (id.length > 0) {
+           url2 = "?stationTypes=" + id
+       }
+       if (term && id.length > 0) {
+           url = '', url2 = ''
+           url3 = "?filterText=" + term + "&stationTypes=" + id
+       }
```

Comment:
I'm glad you shortened this to one call! you can cut down a few lines here by doing something like 
```
let url = '?';
if (term) { url += 'filterText=' + term;
if (id.length > 0) { url += 'stationTypes=' + id;
```
then you can cut out `url2` and `url3`. Also, it's ok to have something like `stationTypes=` (blank) in your http request urls. It'll still work, so you could honestly just try `const url = '?filterText=' + term + '&stationTypes=' + id.join(',')` or something like that 

Also - it's good practice to end each line with a `;`

Reply:
Used the `const url` line that you suggested. `term` returns 'undefined' when a station type is not selected, so I added an if/else statement to fix this.